frontend/app/src/components/calendar/CalendarReminder.vue

```diff
@@ -0,0 +1,313 @@
+      if (divWrapper) {
+        const input = divWrapper.getElementsByTagName('input');
+        if (input[0])
+          input[0].select();
+      }
+    });
+  }
+  else {
+    await addCalendarReminderHandler([
+      {
+        eventId: item.identifier,
+        secsBefore,
+      },
+    ]);
+    await refreshTemporaryData();
+  }
+}
+
+function toggleReminder() {
+  if (get(temporaryData).length > 0)
+    set(showReminders, !get(showReminders));
+}
+
+async function deleteData(index: number) {
+  const item = get(editableItem);
+  const temp = [...get(temporaryData)];
+  const data = temp[index];
+
+  if (!data.isTemporary && item) {
+    try {
+      await deleteCalendarReminder(data.identifier);
+    }
+    catch (error: any) {
+      logger.error(error);
+      notify({
+        display: true,
+        title: t('calendar.reminder.delete_error.title'),
+        message: t('calendar.reminder.delete_error.message', {
+          message: error.message,
+        }),
+      });
+    }
+    await refreshTemporaryData();
+  }
+  else {
+    temp.splice(index, 1);
+    set(temporaryData, temp);
+  }
+}
+
+async function updateData(index: number, value: number) {
+  const item = get(editableItem);
+  const temp = [...get(temporaryData)];
+  const data = temp[index];
+
+  if (item) {
+    if (!data.isTemporary) {
+      try {
+        await editCalendarReminder({
+          identifier: data.identifier,
+          secsBefore: value,
+          eventId: item.identifier,
+        });
+      }
+      catch (error: any) {
+        logger.error(error);
+        notify({
+          display: true,
+          title: t('calendar.reminder.edit_error.title'),
+          message: t('calendar.reminder.edit_error.message', {
+            message: error.message,
+          }),
+        });
+      }
+
+      await refreshTemporaryData();
+    }
+    else if (!isSameSecsBeforeExist(value)) {
+      await addCalendarReminderHandler([
+        {
+          eventId: item.identifier,
+          secsBefore: value,
+        },
+      ]);
+      await deleteData(index);
+      await refreshTemporaryData();
+    }
+  }
+  else {
+    temp[index].secsBefore = value;
+    set(temporaryData, temp);
+  }
+}
+
+watchImmediate(editableItem, refreshTemporaryData);
+
+async function saveTemporaryReminder(eventId: number) {
+  const temporary = get(temporaryData).filter(item => item.isTemporary);
+
+  if (temporary.length > 0) {
+    const savedSeconds = get(temporaryData).filter(item => !item.isTemporary).map(item => item.secsBefore);
+
+    const secsBeforeToSave: number[] = [];
+    temporary.forEach(({ secsBefore }) => {
+      if (!savedSeconds.includes(secsBefore) && !secsBeforeToSave.includes(secsBefore))
+        secsBeforeToSave.push(secsBefore);
+    });
+
+    await addCalendarReminderHandler(secsBeforeToSave.map(secsBefore => ({ secsBefore, eventId })));
+  }
+}
+
+defineExpose({
+  saveTemporaryReminder,
+});
```

Comment:
I wonder if there is an vue eslint rule to move this to at the end.

Reply:
I can't find any 🙁